### ISSUE TITLE
{feature} add prince attack to OCL/CUDA Hashcat

### ISSUE THREAD

**Issue by thesle3p** [hashcat/oclHashcat#56]
Would it be possible to merge PRINCE into OCL Hashcat? Currently I have noticed a bottleneck forming by having PRINCE run on the CPU and communicating with OCL Hashcat using a pipe.

**Comment by epixoip** [comment]
You should be using prince_generated.rule or prince_optimized.rule when piping pp into oclHashcat.

**Comment by jsteube** [comment]
Yep, that would work for now and is the current state of the art of using prince with oclHashcat. 

However, now that we are not more limited by SIMD code in the kernel, we maybe can look again into adding this attack-mode. There's at least a few chances :) Any volunteers for implementing?

**Comment by magnumripper** [comment]
A really fast implementation would have (at least) some inner loop on GPU... but like we discussed before it's certainly not trivial.

**Comment by philsmd** [comment]
This could probably be implemented much easier nowadays since we have the slow candidates (-S) interface... it could also be a nice "template" for other users trying to do similar stuff (see forum posts: https://hashcat.net/forum/thread-8861.html and https://hashcat.net/forum/thread-8571.html etc), prince could be a nice candidate for using the slow candidate way of feeding hashcat with passwords
